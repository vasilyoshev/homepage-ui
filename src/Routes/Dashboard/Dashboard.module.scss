@import url('https://fonts.googleapis.com/css2?family=Montserrat&display=swap');

$headers-font: 'Montserrat', sans-serif;

.dashboardCard {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 360px;
  font-family: $headers-font;
}

.userText {
  margin: auto;
}

h5 {
  margin: 5px 0 0;
  font-size: 1.2rem;
}
