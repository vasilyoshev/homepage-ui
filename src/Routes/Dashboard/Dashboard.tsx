import React from 'react';
import { NavLink } from 'react-router-dom';
import { useAppSelector } from 'store';
import { Card, CardContent, CardActions, Button, Avatar } from '@material-ui/core';
import CloudUploadIcon from '@material-ui/icons/CloudUpload';
import styles from './Dashboard.module.scss';

export const Dashboard: React.FC = () => {
  const date = new Date().toDateString();
  const time = new Date().toLocaleTimeString([], { hour: '2-digit', minute: '2-digit', hour12: false });
  const { username } = useAppSelector((state) => state.user);

  return (
    <Card className={styles.dashboardCard} elevation={6}>
      <CardContent>
        <Avatar className={styles.avatar} style={{ width: '70px', height: '70px' }} alt="avatar" />
        <h2>{`Welcome, ${username}.`}</h2>
        <h5>{date}</h5>
        <h5>{time}</h5>
        <CardActions>
          <Button variant="contained" color="primary">
            <NavLink to="/" activeClassName="selected">
          Go to Widgets
            </NavLink>
          </Button>
          <Button variant="contained" color="default" startIcon={<CloudUploadIcon />}>
          </Button>
        </CardActions>
      </CardContent>
    </Card>
  );
};

