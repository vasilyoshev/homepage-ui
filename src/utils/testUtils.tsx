import React from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';

// TODO extract mocks to separate files, e.g. widgetsMock.js
const mockState = {
  loading: {},
  widgets: { widgetsProps: [] },
  user: {},
};
const createMockStore = configureStore([thunk]);
const store = createMockStore(mockState);

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const Providers = ({ children }: any) => (
  <Provider store={store}>
    <MemoryRouter initialEntries={['/']}>{children}</MemoryRouter>
  </Provider>
);

const customRender = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
  render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';

export { customRender as render };
