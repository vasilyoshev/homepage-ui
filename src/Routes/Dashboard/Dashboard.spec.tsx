import { Dashboard } from './Dashboard';
import { render } from 'utils';

describe('Dashboard', () => {
  it('renders successfully', () => {
    const { asFragment } = render(<Dashboard />);

    expect(asFragment()).toMatchSnapshot();
  });
});
