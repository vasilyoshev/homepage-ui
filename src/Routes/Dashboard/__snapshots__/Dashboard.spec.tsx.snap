// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard renders successfully 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root dashboardCard MuiPaper-elevation6 MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circle avatar MuiAvatar-colorDefault"
        style="width: 70px; height: 70px;"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiAvatar-fallback"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
      <h2>
        Welcome, .
      </h2>
      <h5>
        Fri Jan 01 2021
      </h5>
      <h5>
        2020-04-15
      </h5>
      <div
        class="MuiCardActions-root MuiCardActions-spacing"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <a
              aria-current="page"
              class="selected"
              href="/"
            >
              Go to Widgets
            </a>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM14 13v4h-4v-4H7l5-5 5 5h-3z"
                />
              </svg>
            </span>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
